pennylane/ops/functions/bind_new_parameters.py

```diff
@@ -111,6 +111,30 @@ def bind_new_parameters_composite_op(op: CompositeOp, params: Sequence[TensorLik
     return op.__class__(*new_operands)
 
 
+@bind_new_parameters.register
+def bind_new_parameters_linear_combination(
+    op: qml.ops.LinearCombination, params: Sequence[TensorLike]
+):
+
+    new_coeffs, new_ops = [], []
+    i = 0
+    for o in op.ops:
+        new_coeffs.append(params[i])
+        i += 1
+        if isinstance(o, qml.Hermitian):
+            new_ops.append(qml.Hermitian(params[i], wires=o.wires))
```

Comment:
I have a question, are we sure that the `params` list is in the order assumed here? The coefficient for the `Hermitian` would come before the data of the `Hermitian` itself in the list of `params`?

Reply:
I am assuming the order of how also `op.data` is generated:
```python
op = qml.ops.LinearCombination([0.5, 1.5], [qml.Hermitian(np.eye(2), 0), X(0)])
op.data
```

```
(0.5,
 array([[1., 0.],
        [0., 1.]]),
 1.5)
```

which seems to follow `params = [c0, op0.data, c1, op1.data,...]` where the `opX.data` is optional depending on whether the op carries data or not